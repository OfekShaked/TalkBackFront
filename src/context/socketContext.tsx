import {io} from "socket.io-client";
import React from 'react';
import config from '../configs/config';
const SOCKET_URL = config.apiUrl;

export const socket = io(SOCKET_URL, { transports: ['websocket', 'polling', 'flashsocket'] });
export const SocketContext = React.createContext<any>(null);
