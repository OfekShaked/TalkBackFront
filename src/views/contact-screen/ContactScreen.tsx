import React, { useState } from 'react';
import { DataGrid, GridColDef } from '@material-ui/data-grid';
import ContactDialog from './ContactDialog';


const ContactScreen = () => {

    const [open, setOpen] = useState(false);
    const [selected, setSelected] = useState(null);

    const handleClickOpen = () => {
        setOpen(true);
    };

    const handleClose = (value: string) => {
        setOpen(false);
    };
    const handleRowSelection = (e:any) => {
        handleClickOpen();
        setSelected(e.selectionModel);
    }

    const columns: GridColDef[] = [
        { field: 'id', headerName: 'ID', width: 90 },
        {
            field: 'username',
            headerName: 'User Name',
            width: 150,
            editable: false,
        },
        {
            field: 'status',
            headerName: 'Status',
            width: 150,
            editable: false,
        }
    ];

    return (
        <>
            <DataGrid
                rows={[]}
                columns={columns}
                pageSize={5}
                autoHeight
                disableSelectionOnClick
                onRowClick={handleRowSelection}
            />
            <ContactDialog open={open} onClose={handleClose} />
        </>

    );
}
export default ContactScreen;