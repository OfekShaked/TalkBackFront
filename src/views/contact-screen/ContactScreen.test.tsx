import React from 'react';
import renderer from 'react-test-renderer';
import ContactScreen from './ContactScreen';
import { SocketContext } from '../../context/socketContext';
import { screen, render } from '@testing-library/react'
import SocketMock from 'socket.io-mock';
import { act } from 'react-dom/test-utils';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';
import { DataGrid } from '@material-ui/data-grid';

configure({ adapter: new Adapter() });
let socket = new SocketMock()


describe('contact screen tests', () => {
    it('renders contact screen view', () => {
        const component = renderer.create(
            <ContactScreen ></ContactScreen>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders grid columns', () => {
        const component = mount(
            <SocketContext.Provider value={socket.socketClient}>
                <ContactScreen ></ContactScreen>
            </SocketContext.Provider>
        );
        const contactScreen = component.find(ContactScreen);
        const grid = contactScreen.find(DataGrid);
        expect(grid.get(0).props.columns.length).toBe(3);
    });

    it("renders grid rows", async () => {  
         let component = await mount(
            <SocketContext.Provider value={socket.socketClient}>
                <ContactScreen ></ContactScreen>
            </SocketContext.Provider>
        );   
        await act(async()=>{
            await socket.emit("getUsers", {offlines:[{ username: "ofek123" }, { username: "ofek141" }, { username: "ofek43" }, { username: "o23fek" }, { username: "ofe21k" }],onlines:[{ username: "ofek123" }, { username: "ofek141" }, { username: "ofek43" }, { username: "o23fek" }, { username: "ofe21k" }]});
        })
        await component.update();
        const contactScreen = await component.find(ContactScreen);
        const grid = await contactScreen.find(DataGrid);
        expect(grid.at(0).props().rows.length).toBe(5);
        expect(grid.at(1).props().rows.length).toBe(5);

    })

})

