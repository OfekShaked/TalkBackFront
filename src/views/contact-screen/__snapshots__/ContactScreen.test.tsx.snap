// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact screen tests renders contact screen view 1`] = `
Array [
  <div />,
  <div />,
]
`;
